app/src/main/java/com/poupa/vinylmusicplayer/misc/queue/StaticPlayingQueue.java

```diff
@@ -0,0 +1,458 @@
+
+        songs = new ArrayList<>();
+        songsIsStale = true;
+        resetSongs();
+    }
+
+    public void restoreMode(int shuffleMode, int repeatMode) {
+        this.shuffleMode = shuffleMode;
+        this.repeatMode = repeatMode;
+    }
+
+    /* -------------------- queue modification (add, remove, move, ...) -------------------- */
+
+    // add song at the end of both list
+    public void add(Song song) {
+        queue.add(new IndexedSong((Song)song, queue.size()));
+        originalQueue.add(new IndexedSong((Song)song, originalQueue.size()));
+
+        songsIsStale = true;
+    }
+
+    // add list of song at the end of both list
+    public void addAll(@NonNull List<Song> songs) {
+        for (Song song : songs) {
+            add(song);
+        }
+    }
+
+    // add song after and including position, numbering need to be redone for every song after this position (+1)
+    public void addAfter(int position, Song song) {
+        if (position >= queue.size()) {
+            position = queue.size() - 1;
+        }
+        int previousPosition = queue.get(position).index + 1;
+        position = position + 1;
+
+        originalQueue.add(previousPosition, new IndexedSong(song, previousPosition));
+        queue.add(position, new IndexedSong(song, previousPosition));
+        for (int i = 0; i < queue.size(); i++) {
+            originalQueue.get(i).index = i;
+
+            if (i != position && queue.get(i).index >= previousPosition) {
+                queue.get(i).index = queue.get(i).index + 1;
+            }
+        }
+
+        if (position < this.currentPosition) {
+            this.currentPosition++;
+        }
+
+        songsIsStale = true;
+    }
+
+    public void addSongBackTo(int position, IndexedSong song) {
+        int previousPosition = song.index;
+
+        originalQueue.add(previousPosition, new IndexedSong((Song)song, previousPosition));
+        queue.add(position, new IndexedSong((Song)song, previousPosition));
+        for (int i = 0; i < queue.size(); i++) {
+            originalQueue.get(i).index = i;
+
+            if (i != position && queue.get(i).index >= previousPosition) {
+                queue.get(i).index = queue.get(i).index + 1;
+            }
+        }
+
+        if (position <= this.currentPosition) {
+            this.currentPosition++;
+        }
+
+        songsIsStale = true;
+    }
+
+    // add songs after and including position, numbering need to be redone for every song after this position (+number of song)
+    public void addAllAfter(int position, @NonNull List<Song> songs) {
+        if (position >= queue.size()) {
+            position = queue.size() - 1;
+        }
+        int previousPosition = queue.get(position).index + 1;
+        position = position + 1;
+
+        int n = songs.size() - 1;
+        for (int i = n; i >= 0; i--) {
+            originalQueue.add(previousPosition, new IndexedSong(songs.get(i), previousPosition + i));
```

Comment:
You know where I'm going with `new IndexedSong(songs.get(i), previousPosition + i)` :)